File: server/src/main/java/com/bomberman/factories/LevelFactory.java
package com.bomberman.factories;

import com.bomberman.constants.Configuration;
import com.bomberman.entities.Brick;
import com.bomberman.entities.Position;
import com.bomberman.entities.Wall;
import com.bomberman.game.GameState;

public class LevelFactory {

  private static final int WIDTH = 15;
  private static final int HEIGHT = 11;

  private final int[][] level = // 15x11
      new int[][] {
        {1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0},
        {0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0},
        {0, 0, 0, 0, 0, 0, 0, 2, 2, 2, 0, 0, 0, 0, 0},
        {0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 2, 1, 0},
        {0, 0, 2, 2, 0, 0, 0, 0, 0, 0, 2, 0, 0, 1, 0},
        {0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0},
        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0},
        {0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 2, 1, 0, 1, 0},
        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
        {0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 2, 1, 0, 1, 0},
        {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1}
      };

  public GameState createLevel() {
    final GameState gameState = new GameState();
    for (int y = 0; y < HEIGHT; y++) {
      for (int x = 0; x < WIDTH; x++) {
        final float height = HEIGHT - y - 1;
        final Position position =
            new Position(x * Configuration.TEXTURE_SIZE, height * Configuration.TEXTURE_SIZE);
        switch (level[y][x]) {
          case 1:
            gameState.getWalls().add(new Wall(position));
            break;
          case 2:
            gameState.getBricks().add(new Brick(position));
        }
      }
    }
    return gameState;
  }
}

File: server/src/main/kotlin/com/bomberman/factory/LevelFactory.kt
package com.bomberman.factory

import com.bomberman.TEXTURE_SIZE
import com.bomberman.entity.Position
import com.bomberman.entity.mapobject.Brick
import com.bomberman.entity.mapobject.Wall
import com.bomberman.game.GameState


private const val WIDTH = 15
private const val HEIGHT = 11

class LevelFactory {
  private val level = arrayOf(
      intArrayOf(1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0),
      intArrayOf(0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0),
      intArrayOf(0, 0, 0, 0, 0, 0, 0, 2, 2, 2, 0, 0, 0, 0, 0),
      intArrayOf(0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 2, 1, 0),
      intArrayOf(0, 0, 2, 2, 0, 0, 0, 0, 0, 0, 2, 0, 0, 1, 0),
      intArrayOf(0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0),
      intArrayOf(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0),
      intArrayOf(0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 2, 1, 0, 1, 0),
      intArrayOf(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
      intArrayOf(0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 2, 1, 0, 1, 0),
      intArrayOf(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1))

  fun createLevel(): GameState {
    val gameState = GameState()
    for (y in 0 until HEIGHT) {
      for (x in 0 until WIDTH) {
        val position = Position((x * TEXTURE_SIZE).toFloat(), (y * TEXTURE_SIZE).toFloat())
        when (level[y][x]) {
          1 -> gameState.walls.add(Wall(position))
          2 -> gameState.bricks.add(Brick(position))
        }
      }
    }
    return gameState
  }
}